### PR TITLE
Added AutoFixture 

### PR DESCRIPTION
**Pull request by deivit98**
ArticleControllerTests does not work because of changed logic in the controller

### CODE REVIEW

File: SportsHub/UnitTests/Services/AuthenticationServiceTests.cs [hunk 1]
```diff
@@ -13,36 +17,42 @@ public class AuthenticationServiceTests
...
+        SetupFixture();
+        _userService = _fixture.Freeze<Mock<IUserService>>();
+        _passwordHasher = _fixture.Freeze<Mock<IPasswordHasher>>();
         _authentication = new AuthenticationService(_userService.Object, _passwordHasher.Object);
+        _passwordCheckResult = new PasswordCheckResult(true, false);
     }
 
-    [Fact]
-    public async Task Authenticate_WithUsername_ReturnUser()
+    //Not Working !!!
```
Comment (MBocharskyi) [inline SportsHub/UnitTests/Services/AuthenticationServiceTests.cs:32]: Do you need this comment? Is it true what stated in it?

Comment (deivit98) [reply SportsHub/UnitTests/Services/AuthenticationServiceTests.cs:32]: No its old , probably forgot to delete it with all resolving of conflicts

File: SportsHub/UnitTests/Controllers/ArticleControllerTests.cs [hunk 2]
```diff
@@ -1,134 +1,108 @@
...
             //Arrange
-            string title = "testArticle";
-            var article = ArticleMockData.GetArticle();
+            var article = _fixture.Build<Article>().With(x => x.Title, title).Create();
             _articleService.Setup(service => service.GetByTitleAsync(title)).ReturnsAsync(article);
-
+    
             //Act
             var result = await _articleController.GetArticleByTitleAsync(title);
             var resultObject = GetObjectResultContent<ArticleResponseDTO>(result);
```
[the remark this answers is not in the document]
Comment (deivit98) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs:45]: I did not wrote this and didnt change implementation on this Tests because all of them dont work because of changed logic in the controller

File: SportsHub/UnitTests/Controllers/ArticleControllerTests.cs [hunk 3]
```diff
@@ -1,134 +1,108 @@
...
         private readonly IMapper _mapper;
-
+        private IFixture _fixture;
+    
         public ArticleControllerTests()
         {
+            SetupFixture();
             if (_mapper == null)
             {
                 var mappingConfig = new MapperConfiguration(x => x.CreateMap<Article, ArticleResponseDTO>());
```
[the remark this answers is not in the document]
Comment (deivit98) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs:27]: In both Controllers is like this maybe we can be changed when we create the task to refactor more of the tests because beside this UserServiceTests need another thests for added method and ArticleControllerTests need new dependancy to work

**Comment by deivit98** [comment]
I need to change UserService . The name are still return null but we actually throw exception

File: SportsHub/UnitTests/Controllers/ArticleControllerTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (deivit98) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs]: From what i have seen now the result cannot be cast with as ObjectResult it gave me errors . My colleague implemented a method which returns the objectResult thats why these 2 line are in the ACT probably . What do you want me to do all of this controller tests are not working we need to change them all probably and we have to change other tests which check for null to check for exceptions

**Review by Banovvv — APPROVED** [review]

File: SportsHub/UnitTests/Controllers/ArticleControllerTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MBocharskyi) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs]: It is ok if you change, these tests. My comment was that ` var resultObject = GetObjectResultContent<ArticleResponseDTO>(result);` should be in Assert section.

**Review by MBocharskyi — APPROVED** [review]

**Comment by MBocharskyi** [comment]
Please resolve conflicts.

File: SportsHub/UnitTests/Controllers/ArticleControllerTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (deivit98) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs]: Yea but we use result from ACT to get resultObject

[the remark this answers is not in the document]
Comment (MBocharskyi) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs]: For me, `Act` is a method call of service/repository, etc. All other actions mapping/getting specific model is not part of `Act` they are part of `Assert`.

[the remark this answers is not in the document]
Comment (deivit98) [reply SportsHub/UnitTests/Controllers/ArticleControllerTests.cs]: Yea i have changed it . I combined the two lines in one

**Review by NataliiaBidiuk — APPROVED** [review]
